framework/src/main/java/org/checkerframework/common/basetype/BaseTypeValidator.java

```diff
@@ -185,8 +185,8 @@ protected void reportValidityResultOnUnannotatedType(
             final Tree p) {
         // TODO: if underlying is a compound type such as List<@Palindrome String>, then it would be
         // nice to print it as "List" instead of as "List<@Palindrome String>".
-        TypeMirror underlying = type.getUnderlyingType();
-        checker.report(Result.failure(errorType, type.getAnnotations(), underlying.toString()), p);
+        TypeMirror unannotated = TypeAnnotationUtils.unannotatedType(type.getUnderlyingType());
```

Comment:
While you're at it, can you try to address the TODO comment above?
Write two extra tests:
1. the top-level type has invalid annotations: `@Bad List<String>` and we only want to see `List`
2. the type argument has invalid annotations: `List<@Bad Map<...>>` and we only want to see `Map`

To get to the erase type, try changing `type.getUnderlyingType()` to `type.getErased().getUnderlyingType()`.

Reply:
No problem, I will do it.

Reply:
Friendly ping :-)

Reply:
Sorry for the delayed response, I have already updated the PR.
I have kept thinking about how to create these two extra test cases, I tried some test cases but failed to show the differences. I am still confused about it... Could you please give me some advises?